coordinator/ethereum/message-anchoring-i2/src/test/kotlin/linea/anchoring/MessageAnchoringAppTest.kt

```diff
@@ -231,8 +231,26 @@ class MessageAnchoringAppTest {
 
     addLogsToFakeEthClient(ethLogs)
     l1Client.setFinalizedBlockTag(ethLogs.last().messageSent.log.blockNumber + 10UL)
+    l2MessageService.forceAnchoringFailures = true
 
     anchoringApp.start().get()
+    // wait for the queue to fill up
+    await()
+      .atMost(10.seconds.toJavaDuration())
+      .untilAsserted {
+        assertThat(anchoringApp.eventsQueueSize).isGreaterThan(0)
+      }
+    // ensure it has multiple event polling ticks but does
+    // not fetch more events that those already fetched
+    // in the 1st tick that filled the queue
+    await()
+      .atMost(2.seconds.toJavaDuration())
+      .untilAsserted {
+        assertThat(anchoringApp.eventsQueueSize).isBetween(1, 100)
```

Comment:
Why 1 to 100? the max should be `messageQueueCapacity` which is 20, no?

Reply:
No not really. 20 is supposed to be a soft limit of how many messages can be in memory at a time, but more than 20 events are supposed to be achored with multiple iterations

Reply:
I think `messagesToFetch` name is inaccurate, because what this argument actually means is when log collection should stop. I changed the name of that arg and modified verifications a bit so they are a bit stricter

Reply:
Thanks for change, LGTM @Filter94, but can we say that the size of `eventsQueue`, according to how we use it here, would always be  0 to `messageQueueCapacity`? Just to make sure my understanding is correct, cause I can see a case here where the size of `eventsQueue` would go over `messageQueueCapacity`

Reply:
No we can't. It's at most `messageQueueCapacity` + max number of messageSent events in 1 block

Reply:
> It's at most messageQueueCapacity + max number of messageSent events in 1 block
This is where I'm confused, don't we already capped the `logsSoftLimit` as `l1MessagesSentFetchLimit.coerceAtMost(remainingCapacity.toUInt())` at `L1MessageSentEventsPoller line70` for each tick?

Reply:
We did, but I'm not following. When you say "already" what other place in the code do you mean where we're capping it again?

Reply:
oh now I see, `eventsFetcher.findL1MessageSentEvents()` could return number of `MessageSentEvents` events more than `messagesToFetch` and that's why you renamed it as `logsSoftLimit`, now it makes sense to me.